### PR TITLE
Setting user-agent header to dask-gcsfs/<version>

### PR DESCRIPTION
**Pull request by vlasenkoalexey**
@sshrdp FYI

### CODE REVIEW

File: gcsfs/core.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import re
 import requests
 import time
+import versioneer
```
Comment (martindurant) [inline gcsfs/core.py:28]: `from .. import version`

Comment (martindurant) [reply gcsfs/core.py:28]: (might need to change order of inputs in `__init__`)

File: gcsfs/core.py [hunk 2]
```diff
@@ -440,6 +442,11 @@ def _call(self, method, path, *args, **kwargs):
         data = kwargs.pop("data", None)
         r = None
 
+        if headers is None:
+            headers = {}
+        if "User-Agent" not in headers:
+            headers["User-Agent"] = "dask-gcsfs/" + VERSION
```
Comment (martindurant) [inline gcsfs/core.py:448]: I suggest `"python-gcsfs/"`, since this is not necessarily used in conjunction with Dask.

**Comment by vlasenkoalexey** [comment]
Thanks for quick reply, I'll resolve your comments.
Unfortunately I wasn't able to make tests to pass, see https://github.com/dask/gcsfs/issues/251

**Comment by vlasenkoalexey** [comment]
Have to rebase, will send another PR for this

**Pull request closed without merging**